src/main/java/seedu/taassist/model/ModelManager.java

```diff
@@ -114,6 +114,27 @@ public void setStudent(Student target, Student editedStudent) {
         taAssist.setStudent(target, editedStudent);
     }
 
+    @Override
+    public boolean hasModuleClass(ModuleClass moduleClass) {
+        requireAllNonNull(moduleClass);
+        return taAssist.hasModuleClass(moduleClass);
+    }
+
+    @Override
+    public void deleteModuleClass(ModuleClass target) {
+        taAssist.removeModuleClass(target);
+    }
+
+    @Override
+    public void addModuleClass(ModuleClass moduleClass) {
```

Comment:
Would it be better to still check for `requireNonNull` for defensive coding?

Reply:
Ok! Great suggestion! I fixed this for some of the original AB3 methods too.